components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt

```diff
@@ -113,6 +120,56 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
     }
 }
 
+/**
+ * Shares content via [ACTION_SEND] intent.
+ *
+ * @param text the data to be shared [EXTRA_TEXT]
+ * @param subject of the intent [EXTRA_TEXT]
+ * @return true it is able to share false otherwise.
+ */
+fun Context.shareMedia(
+    filePath: String,
+    contentType: String?,
+    subject: String? = null,
+    message: String? = null
+): Boolean {
+    val contentUri = FileProvider.getUriForFile(
+        this,
+        "${applicationContext.packageName}.feature.downloads.fileprovider", // (packageName + FILE_PROVIDER_EXTENSION)
+        File(filePath)
+    )
+
+    val intent = Intent().apply {
+        action = ACTION_SEND
+        type = contentType ?: contentResolver.getType(contentUri)
+        flags = FLAG_ACTIVITY_NEW_DOCUMENT or FLAG_GRANT_READ_URI_PERMISSION
+        putExtra(EXTRA_STREAM, contentUri)
+        if (subject != null) {
+            putExtra(EXTRA_SUBJECT, subject)
+        }
+        if (message != null) {
+            putExtra(EXTRA_TEXT, message)
+        }
+    }
+
+    val shareIntent = Intent.createChooser(intent, getString(R.string.mozac_support_ktx_menu_share_with)).apply {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            // Android Q allows us to show a thumbnail preview of the file to be shared.
+            clipData = ClipData.newRawUri(contentUri.toString(), contentUri)
+        }
+
+        flags = FLAG_ACTIVITY_NEW_TASK or FLAG_GRANT_READ_URI_PERMISSION
+    }
+
+    return try {
+        startActivity(shareIntent)
+        true
+    } catch (error: ActivityNotFoundException) {
+        Log.log(Log.Priority.WARN, message = "No activity to share to found", throwable = error, tag = "shareMedia")
+        false
```

Comment:
Could add this for this line? :)

Reply:
The `true` return value is already tested.
Will add a test for the `false` value also.
Using mockito's throw/answer to force the exception for `startActivity()`methods wouldn't want to work so after many tries making this work I resolved to using `FakeContext` with an overridden `startActivity`. Hope it's ok.

Reply:
In case you need an additional reference we have similar test in [AppLinks](https://github.com/mozilla-mobile/android-components/blob/9673de174eefa9fa957c351877c3ea3de3de5caf/components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksUseCasesTest.kt#L366)